Compilation warnings

Getting these warnings with version 2.1.1 on FreeBSD 10 with clang 3.4.1:

```
/usr/ports/net/zmap/work/zmap-2.1.1/lib/util.c:252:6: warning: implicit declaration of function 'pthread_setaffinity_np' is invalid in C99 [-Wimplicit-function-declaration]
        if (pthread_setaffinity_np(pthread_self(),
            ^
/usr/ports/net/zmap/work/zmap-2.1.1/src/ztee.c:218:6: warning: implicit declaration of function 'getline' is invalid in C99 [-Wimplicit-function-declaration]
        if (getline(&first_line, &first_line_len, stdin) < 0) {
            ^
In file included from /usr/ports/net/zmap/work/zmap-2.1.1/src/get_gateway.c:27:
In file included from /usr/ports/net/zmap/work/zmap-2.1.1/src/get_gateway-bsd.h:23:
/usr/local/include/dnet/ip.h:349:4: warning: 'opt_data' may not be nested in a struct due to flexible array member [-Wflexible-array-extensions]
        } opt_data;
          ^
```

Still warnings, even more:
```
In file included from /usr/local/include/unitypes.h:22:
/usr/local/include/unistring/stdint.h:3:5: warning: '__GLIBC__' is not defined, evaluates to 0 [-Wundef]
#if __GLIBC__ >= 2
    ^
In file included from /home/yuri/github/zmap/src/fieldset.c:15:
/usr/local/include/unistr.h:204:7: warning: 'CONFIG_UNICODE_SAFETY' is not defined, evaluates to 0 [-Wundef]
#  if CONFIG_UNICODE_SAFETY
      ^
/usr/local/include/unistr.h:208:7: warning: 'CONFIG_UNICODE_SAFETY' is not defined, evaluates to 0 [-Wundef]
#  if CONFIG_UNICODE_SAFETY
      ^
3 warnings generated.
In file included from /home/yuri/github/zmap/src/get_gateway.c:27:
/home/yuri/github/zmap/src/get_gateway-bsd.h:24:7: warning: '__GNUC_MINOR_' is not defined, evaluates to 0 [-Wundef]
#elif __GNUC_MINOR_ >= 6
      ^
In file included from /home/yuri/github/zmap/src/get_gateway.c:27:
In file included from /home/yuri/github/zmap/src/get_gateway-bsd.h:30:
/usr/local/include/dnet/ip.h:349:4: warning: 'opt_data' may not be nested in a struct due to flexible array member [-Wflexible-array-extensions]
        } opt_data;
          ^
2 warnings generated.
In file included from /home/yuri/github/zmap/src/fieldset.c:15:
In file included from /usr/local/include/unistr.h:21:
In file included from /usr/local/include/unitypes.h:22:
/usr/local/include/unistring/stdint.h:3:5: warning: '__GLIBC__' is not defined, evaluates to 0 [-Wundef]
#if __GLIBC__ >= 2
    ^
In file included from /home/yuri/github/zmap/src/fieldset.c:15:
/usr/local/include/unistr.h:204:7: warning: 'CONFIG_UNICODE_SAFETY' is not defined, evaluates to 0 [-Wundef]
#  if CONFIG_UNICODE_SAFETY
      ^
/usr/local/include/unistr.h:208:7: warning: 'CONFIG_UNICODE_SAFETY' is not defined, evaluates to 0 [-Wundef]
#  if CONFIG_UNICODE_SAFETY
      ^
3 warnings generated.
In file included from /home/yuri/github/zmap/src/get_gateway.c:27:
/home/yuri/github/zmap/src/get_gateway-bsd.h:24:7: warning: '__GNUC_MINOR_' is not defined, evaluates to 0 [-Wundef]
#elif __GNUC_MINOR_ >= 6
      ^
In file included from /home/yuri/github/zmap/src/get_gateway.c:27:
In file included from /home/yuri/github/zmap/src/get_gateway-bsd.h:30:
/usr/local/include/dnet/ip.h:349:4: warning: 'opt_data' may not be nested in a struct due to flexible array member [-Wflexible-array-extensions]
        } opt_data;
          ^
2 warnings generated.
```

--- a/src/get_gateway-bsd.h
+++ b/src/get_gateway-bsd.h
@@ -17,7 +17,7 @@
 #include <net/if.h>
 #include <net/if_dl.h>
 
-#if !defined(__APPLE__)
+#if defined(_SYSTYPE_BSD)
 
 #if __GNUC__ < 4
 #error "gcc version >= 4 is required"

